### PR TITLE
Extract total comand logic

### LINKED ISSUE #217: Extract total command logic to proper handler
## Task Context
As a next steps when working on project refactoring, we decided to extract the code that is handling specific commands to separate classes. 
This task is about extracting current `total` command logic from `HydrateReminderPlugin` class to `TotalCommandHandler ` class.

## Acceptance Criteria / Goals
- `total` command logic is extracted to dedicated class `TotalCommandHandler`
- `HydrateReminderPlugin::onCommandExecuted` should call `commandDelegate.invokeCommand(commandExecuted)` when the selected command is `total`
- Solution was tested using unit tests

## Additional Notes
Please refer to [CONTRIBUTING.md](https://github.com/jmakhack/hydrate-reminder/blob/master/CONTRIBUTING.md) for info on how to setup, build, run, and test this project. Feel free to ping @jmakhack or @mslowiak in the comments below with any questions if needed. Also, join the [Discord](https://discord.gg/RTgxfFW9mS) to meet and engage with other contributors!

[](https://discord.gg/RTgxfFW9mS)

### PR DESCRIPTION
**Pull request by harikesh409**
## Associated Issue

Closes #217 

## Implemented Solution

Refactored code as suggested in issue

## Testing Details

Operating System: MacOS Monterey 
RuneLite Version: 1.8.34.1

### CODE REVIEW

**Review by jmakhack — COMMENTED** [review]
@harikesh409 your changes all look good! I'll test the changes when I have time and then approve this if it all works as expected.

Until then I've marked this pull request as hacktoberfest-approved :)

**Comment by harikesh409** [comment]
> @harikesh409 your changes all look good! I'll test the changes when I have time and then approve this if it all works as expected.
> 
> Until then I've marked this pull request as hacktoberfest-approved :)

Awesome thank you 😊

File: src/main/java/com/hydratereminder/HydrateReminderPlugin.java [hunk 1]
```diff
@@ -254,7 +255,7 @@ public void onCommandExecuted(CommandExecuted commandExecuted)
 			}
 		}
 		// TODO: Uncomment when the commands will be refactored to com.hydratereminder.command
-		// commandDelegate.invokeCommand(commandExecuted);
+//		 commandDelegate.invokeCommand(commandExecuted);
```
Comment (mslowiak) [inline src/main/java/com/hydratereminder/HydrateReminderPlugin.java:258]: This change is not needed in scope of this PR

File: src/test/java/com/hydratereminder/command/total/TotalCommandHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    HydrateReminderPlugin hydrateReminderPlugin;
+    @InjectMocks
+    TotalCommandHandler totalCommandHandler;
+
+    @Test
+    public void shouldSendHydrate() {
+        totalCommandHandler.handle();
+
+        verify(hydrateReminderPlugin,times(1)).getCurrentSessionHydrationBreaks();
+        verify(chatMessageSender,times(1)).sendHydrateEmojiChatGameMessage(anyString());
```
Comment (mslowiak) [inline src/test/java/com/hydratereminder/command/total/TotalCommandHandlerTest.java:30]: Can you test it providing exact string that should be tested? `anyString()` is too broad.
There is a difference `break` vs `breaks` in the logic. Please cover that in tests

**Review by mslowiak — CHANGES_REQUESTED** [review]

File: src/main/java/com/hydratereminder/command/total/TotalCommandHandler.java [hunk 3]
```diff
@@ -1,15 +1,34 @@
...
+import javax.inject.Inject;
 import javax.inject.Singleton;
 
 import com.hydratereminder.command.CommandHandler;
 
 @Singleton
 public class TotalCommandHandler implements CommandHandler {
 
+    @Inject
+    private ChatMessageSender chatMessageSender;
```
Comment (mslowiak) [inline src/main/java/com/hydratereminder/command/total/TotalCommandHandler.java:14]: You can make `chatMessageSender` and `hydrateReminderPlugin` final, and then use Inject on constructor rather on field level

File: src/test/java/com/hydratereminder/command/total/TotalCommandHandlerTest.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@RunWith(MockitoJUnitRunner.class)
+public class TotalCommandHandlerTest {
+
+    @Mock
+    ChatMessageSender chatMessageSender;
```
Comment (mslowiak) [inline src/test/java/com/hydratereminder/command/total/TotalCommandHandlerTest.java:19]: all of those variables can be private

File: src/main/java/com/hydratereminder/HydrateReminderPlugin.java (hunk 1 above)
Comment (harikesh409) [reply src/main/java/com/hydratereminder/HydrateReminderPlugin.java]: Reverted the change

File: src/test/java/com/hydratereminder/command/total/TotalCommandHandlerTest.java (hunk 2 above)
Comment (harikesh409) [reply src/test/java/com/hydratereminder/command/total/TotalCommandHandlerTest.java]: Updated and added additional tests as suggested.

File: src/main/java/com/hydratereminder/command/total/TotalCommandHandler.java (hunk 3 above)
Comment (harikesh409) [reply src/main/java/com/hydratereminder/command/total/TotalCommandHandler.java]: Updated

File: src/test/java/com/hydratereminder/command/total/TotalCommandHandlerTest.java (hunk 4 above)
Comment (harikesh409) [reply src/test/java/com/hydratereminder/command/total/TotalCommandHandlerTest.java]: Updated

File: src/test/java/com/hydratereminder/command/total/TotalCommandHandlerTest.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+import com.hydratereminder.HydrateReminderPlugin;
+import com.hydratereminder.chat.ChatMessageSender;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import static org.mockito.ArgumentMatchers.anyString;
```
Comment (mslowiak) [inline src/test/java/com/hydratereminder/command/total/TotalCommandHandlerTest.java:11]: codacy is failing on unused import

File: src/main/java/com/hydratereminder/command/total/TotalCommandHandler.java [hunk 6]
```diff
@@ -1,15 +1,36 @@
...
 public class TotalCommandHandler implements CommandHandler {
 
+    private final ChatMessageSender chatMessageSender;
+
+    private final HydrateReminderPlugin hydrateReminderPlugin;
+
+    public TotalCommandHandler(ChatMessageSender chatMessageSender, HydrateReminderPlugin hydrateReminderPlugin) {
+        this.chatMessageSender = chatMessageSender;
+        this.hydrateReminderPlugin = hydrateReminderPlugin;
+    }
```
Comment (macslowiak) [inline src/main/java/com/hydratereminder/command/total/TotalCommandHandler.java:19]: Injection won't be created. Missing @Inject annotation over constructor.
When trying to run: 
image

**Review by macslowiak — CHANGES_REQUESTED** [review]

File: src/main/java/com/hydratereminder/command/total/TotalCommandHandler.java (hunk 6 above)
Comment (harikesh409) [reply src/main/java/com/hydratereminder/command/total/TotalCommandHandler.java]: Added the missing `@Inject` annotation.

File: src/test/java/com/hydratereminder/command/total/TotalCommandHandlerTest.java (hunk 5 above)
Comment (harikesh409) [reply src/test/java/com/hydratereminder/command/total/TotalCommandHandlerTest.java]: Optimized the imports

**Review by macslowiak — APPROVED** [review]

**Review by jmakhack — APPROVED** [review]

**Comment by jmakhack** [comment]
@all-contributors please add @harikesh409 for code and tests

**Pull request merged**